### DISCUSSION TITLE
Potentiometer calibration and range

### DISCUSSION THREAD

**Discussion by tigert** [MobiFlight/MobiFlight-Connector#512] [Ideas]
So potentiometers have some current issues on MSFS2020.

- They only work with custom inputs on msfs2020, unless you use fsuipc, since msfs2020 events cannot take a parameter. So if 
  you add them to hubhop, they should not be included in events.txt unless MF knows how to handle those.
- Potentiometer ranges cannot be calibrated in MobiFlight yet, so you need to adjust that also in the custom code field.
  As the pot actual range probably depends on the component, it does not make sense to put it in the custom RPN code 
  that you share with others in hubhop.
- Potentiometer things in the sim usually work in 0-100% range, so you always need to adjust for this in MobiFlight as in
  Mobiflight they are 0-1023. If potentiometers also represented their values as percentage (0-100) the mapping would be 
  more straightforward, but the internal representation of everything as positive integer might reduce the precision of the values,
  which would not be good.

Lets figure out what practical issues there might be to implement calibration for pots, and to make potentiometer-aware events possible to share in hubhop.

**Comment by pizman** [comment]
Basically good ideas !

On Point 3 i see no Problems with Values of 0-100 !
Usage of MF Potis ( as you request months ago) are things like Dimm Selectors and so on.
Whatever if you move the Poti just a little bit this means the Poti goes maybe 50 from 1024  "Steps" forward.     ( so ~~20 Movements from Min to Max) 
So a value range from 0-100 is still 2 times more as we would be able to effectivltly do by our hands.
Ask yourself.... If you increase Brightness of a LED..... You will not see a difference if you increase by 1% .... May 5% ( again a 1/20 Move) is the point you will see a change of bright. 
So i think 0-100 is pretty sure not 0-64000 like a Windows Joystick..... But still enough fore our usecases.

**LAST NOTE: !!!**
On that point we must also talk about the new "HID Joystick Inputs" in MF ( Beta 9.005 above)
If this tool get implement and work fine we no longer are bind to Poti on the Mega.
If i understand right then we could use the Windows Calibration, too.    So the Data MF reccive from the Potis (Joy Axis) are still perfect.

Please not missunderstand.... I not vote to remove the Poti support on the Mega ( internal MF)  But i ask myself if it make sense to work on that tool ( e.g. implement a Calibration) if we NOW be able to use Potis with a cheap Arduino Micro as a HID Device already within Mobiflight !
Badly Sebastian not do this 6 Months ago..... I think then we would never planed to put a Poti on the Mega itself !

**Comment by tigert** [comment]
Yeah, doing HID joystick and using the inputs in MF is possible now.

Direct potentiometer support in MobiFlight is useful though for panels that mix potentiometers with output leds. As output leds do not work in HID devices, this would then require having two usb devices in a panel, one for pots (MF arduino or HID joystick) and another for leds (MF arduino). 

![image](https://user-images.githubusercontent.com/2587818/136330730-21312db6-767c-4424-853b-505d6eb2725a.png)

Another use case would be for example the TBM930 throttle that I would probably wish to implement using MobiFlight, and while I havent thought about it much, I would guess some microswitches would be required to detect the different modes and positions:

![image](https://user-images.githubusercontent.com/2587818/136330548-b3527690-ec49-4221-8701-b556ae70a093.png)

Though I know the throttle could be a usb device, and the logic could still be done in MobiFlight. But panels can also use other facilities from MobiFlight like pwm control for backlight leds which do not work in HID devices. 

These are both rather "slow" input devices, so the input frequency does not even need to be super frequent.

And yes, I know the way I build panels as stand-alone units that just plug to USB and maybe 12V is different from the traditional way of building a cockpit where you just wire all the backlights in a LED controller etc..
